### ISSUE TITLE
Potential Bug in CUDA implementation of matrix_set_diag on Windows

### ISSUE THREAD

**Issue by ed-alertedh** [tensorflow/tensorflow#13335]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution**: Windows 10 (64-bit)
- **TensorFlow installed from**: binary
- **TensorFlow version (use command below)**: b'unknown' 1.3.0
- **Python version**: 3.6.2
- **CUDA/cuDNN version**: CUDA 8.0 /  cuDNN 6.0
- **GPU model and memory**: Nvidia GTX 1060 6GB

### Describe the problem
I ran into this while trying to run the code at https://github.com/deepmind/dnc. It seems that tf.matrix_set_diag can cause tensorflow to crash when run on GPU under Windows. I am able to run other complex models on GPU so it seems as though my environment is configured correctly for CUDA.
### Source code / logs
```
import tensorflow as tf

batch_size = 16
num_writes = 1
memory_size = 16

with tf.device('/gpu:0'):
    link = tf.Variable(tf.ones([batch_size, num_writes, memory_size, memory_size], dtype=tf.float32))
    test_op = tf.matrix_set_diag(link,tf.zeros([batch_size, num_writes, memory_size],dtype=tf.float32))

sess = tf.Session()
sess.run(tf.global_variables_initializer())
print(sess.run(test_op))
```
Error log here for normal execution:
[error_log.txt](https://github.com/tensorflow/tensorflow/files/1336102/error_log.txt)

The log from running with cuda-memcheck shows what looks like an issue with a null pointer buried deep inside Eigen...
[error_log_cuda_memcheck.txt](https://github.com/tensorflow/tensorflow/files/1336103/error_log_cuda_memcheck.txt)

**Comment by poxvoculi** [comment]
@rmlarsen Possible bug in an eigen-compiled GPU op on windows: is this expected to work without a local compile from source?

**Comment by ed-alertedh** [comment]
Happy to try building from source if you think it will help! It occurs to me that I updated my CUDA Toolkit to 8.0.61_win10 Patch 2 recently. Any chance that is the problem? I wouldn't expect the updated dlls to be incompatible with code linked against the old dlls, but the way MSVC's linker works may as well be voodoo to me...

**Comment by gunan** [comment]
This is fixed at head now.

**Comment by ed-alertedh** [comment]
Thanks! If anyone else has run into this, the changes were merged in #13677 which happened just after 1.4.0rc0 was released
